lib/static-files-router.js

```diff
@@ -0,0 +1,287 @@
+function StaticRouter() {
+    var log = new Log("[static-file-router]");
+    var knownMimeTypes = {
+        // text
+        txt: 'text/plain',
+        html: 'text/html',
+        htm: 'text/html',
+        js: 'application/x-javascript',
+        css: 'text/css',
+        xml: 'application/xml',
+        hbs: 'text/x-handlebars-template',
+        // fonts
+        woff: 'application/font-woff',
+        otf: 'application/font-sfnt',
+        ttf: 'application/font-sfnt',
+        // images
+        jpg: 'image/jpeg',
+        jpeg: 'image/jpeg',
+        png: 'image/png',
+        bmp: 'image/bmp',
+        gif: 'image/gif',
+        svg: 'image/svg+xml'
+    };
+
+    /**
+     * Splits a full file name to its name and extension.
+     * @param fullFileName {string} file name to be split e.g. foo.txt
+     * @return {{name: string, extension: string}} splited parts
+     */
+    function splitFileName(fullFileName) {
+        var index = fullFileName.lastIndexOf(".");
+        return {name: fullFileName.substr(0, index), extension: fullFileName.substr(index + 1)}
+    }
+
+    /**
+     * Returns a file object that represent the specified file.
+     * @param unitFullName {string} full name of the unit
+     * @param resourceType {string} resource type
+     * @param fileName {string} file name with extension
+     * @param lookUpTable {Object} lookup table
+     * @param configs {Object} configurations
+     * @return {Object|null} if file exists a File object, otherwise <code>null</code>
+     */
+    function getFile(unitFullName, resourceType, fileName, lookUpTable, configs) {
+        var unitModel = lookUpTable.units[unitFullName];
+        if (!unitModel) {
+            return null; // unit does not exists
+        }
+        var filePathRelativeToUnit = null;
+        if (resourceType == "less") {
+            filePathRelativeToUnit = configs.constants.DIRECTORY_APP_UNIT_PUBLIC + "/less/"
+                                     + splitFileName(fileName).name + ".less";
+        } else {
+            filePathRelativeToUnit = configs.constants.DIRECTORY_APP_UNIT_PUBLIC + "/"
+                                     + resourceType + "/" + fileName;
+        }
+        // search in this unit
+        var filePath = configs.constants.DIRECTORY_APP_UNITS + "/" + unitFullName
+                       + filePathRelativeToUnit;
+        var file = new File(filePath);
+        if (file.isExists() && !file.isDirectory()) {
+            return file; // this unit has the file
+        }
+        // search in parent unit
+        var parentUnitFullName = unitModel.definition["extends"];
```

Comment:
Related to the 'unit inheritance chaining' feature. Will be implemented in a next PR.